### ISSUE TITLE
Feature request: ability to show a point for each value

### ISSUE THREAD

**Issue by synw** [chrisburnell/spark-line#1]
## Expected behaviour

Add the ability to draw a point for each datapoint, not only the last one. Ideally the point could have a tooltip showing the value on hover

[Edit] : a possible variation would be to to have the min and max value points, like this:

![single_number](https://user-images.githubusercontent.com/12034394/169040637-6b2d1ec9-1079-45ed-a795-62cf918a28c9.png)

## Actual behaviour

We can only add the last point

@chrisburnell

**Comment by chrisburnell** [comment]
Hey @synw,

Thanks for the feature request! This is a really interesting idea. While I'd like to keep spark-line as minimal and simple as possible, I had a crack at adding support for multiple points and multiple colours, and I think it works and looks pretty good!

Give me a bit of time to make sure I'm happy with the code before I publish it, but in the meantime, you can play with what I've got so far on CodePen:

https://codepen.io/chrisburnell/pen/GRQmNqo/2a28dfa7748a7876cf2d6f0ed2ecd9bd

Thanks,
Chris

**Comment by synw** [comment]
Hi. Thanks for your work on this idea. I checked out the code: the end result is just fine but the api may be improved imho. I would prefer something more simple than having to declare lists of values for `points` and `colors`. Maybe to get all points with a default color we could use a boolean flag: like:

```html
<spark-line values="7,0,0,8,0,12,0,6" drawpoints="true" pointcolor="steelblue"></spark-line>
```
This with the option of providing a point color (or use default if not provided) would be enough for me. About the min / max I suggest a similar api with just a simple flag:

```html
<spark-line values="7,0,0,8,0,12,0,6" minmax="true"></spark-line>
```
If this flag is provided we can just get the min/max from the values (something like `Math.max(...valuesList)`) and draw the min max points with default colors, or custom colors if provided

Do you know the old [Jquery sparklines](https://omnipotent.net/jquery.sparkline/#s-about) library? I used it in the past and it is my reference for sparklines. Please check their examples to see the "display the value in a tooltip on hover" feature I mentioned above

**Comment by chrisburnell** [comment]
Hi @synw,

I've published [v1.1.2](https://github.com/chrisburnell/spark-line/releases/tag/1.1.2) which adds two new attributes to `<spark-line>`:

1. `points`: supercedes `endpoint`; comma-delimited string of integers representing at which pairing `values` you want points to appear at; arrays of a length less than the length of the `values` array will be looped over according to `values`
2. `colors`: supercedes `endpoint-color`; comma-delimited string of integers representing the colour of the paired `points`; arrays of a length less than the length of the `values` array will be looped over according to `values`

I've gone into more detail on the [documentation on my website](https://chrisburnell.com/spark-line/). These additions should allow for enough flexibility and control to draw and colour your min/max values as well as any many other variations. You should be able to combine `Math.max()` and `Math.min()` to find the indexes of the min/max values in your `values` array, and from that information generate a `points` array to tell `<spark-line>` at which values to print your points. And a similar process should work for setting the colours too.

As I mentioned before, I'd like to keep this project quite lean, so I've decided against adding any other features from jQuery Sparklines, but thanks very much again for submitting this feature request!

Thanks,
Chris
